department-of-veterans-affairs/abd-vro
# Change bgs rabbitmq configs for svc-bgs-api

## What was the problem?

BGS RabbitMQ Queue and Exchange configurations need to be updated to follow rules/guidelines set in #3329 

Associated tickets or Slack threads:
- #3329
- This is only part of the work to do in #3330 

## How does this fix it?[^1]

The following changes were made to follow the rules below:
* Modified exchange `bgs-api` to `svc-bgs-api.requests` and set auto-delete=false
* Added `svc_bgs_api.healthcheck` exchange
* Modified queue `healthcheck` to `svc-bgs-api.healthcheck` and bound to newly created `svc_bgs_api.healthcheck` exchange
* Modified queue `healthcheck-reply` to have autogenerated name, durable=false, exclusive=True, autodelete=true
  * This autogenerated queue is exclusive and autodelete and created in the healthcheck.rb script which means it likely will not exist long enough to show up in console.
* Modified `add-note` to `svc_bgs_api.add_note` and changed policy to `svc_bgs_api.queues`

## How to test this PR
Passing CI 


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
